tests/integration/test_pip.py

```diff
@@ -136,6 +175,27 @@ def test_pip_packages(
             ["registry.fedoraproject.org/fedora-minimal:37"],
             id="pip_e2e_test",
         ),
+        # Same as above, but with wheel distributions - allow binary option is set to true
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachi2-pip-wheels",
+                ref="b3dea71a1ad51da90e1161cf71c85e0f7c3e292b",
+                packages=(
+                    {
+                        "type": "pip",
+                        "requirements_files": ["requirements.txt"],
+                        "requirements_build_files": ["requirements-build.txt"],
+                        "allow_binary": "true",
```

Comment:
I wonder if `allow_binary` being the only major change justifies a dedicated e2e test instead of enhancing the existing one to pull down wheels as well.

Reply:
I think there's value in making sure that `"allow_binary": "false",` works as intended... but maybe there's another way to do it :shrug:

Reply:
We did that in the isolated scenarios, this is full-blown e2e test which can be tested in a complex manner IMO. I'm not objecting, I'm just not convinced, that's all.

Reply:
Is it related to the question I asked above that it is worth having two scenarios when the build will succeed if `allow_binary` is set to `true` otherwise, the build will fail because of missing `requirements-build.txt` ?

Just to make sure, for users that do not want to specify build dependencies it works

Reply:
> or write an app that depends on a package which only distributes as a wheel.

This covers another test in the repo, but it is not e2e.

Reply:
>> or write an app that depends on a package which only distributes as a wheel.

> This covers another test in the repo, but it is not e2e.

I'm trying to understand where this all got stuck. The fact that we have a fetch test for a case with no sdists doesn't mean we shouldn't test e2e in a complex manner, IOW include e.g. tensorflow or pycaret in requirements for the test app and try building it.

Reply:
So, can I drop the e2e branch altogether and make no-sdists as the e2e test even if there is no _requirements_build.txt_ file?